.github/workflows/pipeline.yml

```diff
@@ -28,68 +28,55 @@ jobs:
-        run: |
-          kubelogin convert-kubeconfig -l azurecli
-
-      # Verify kubectl works
-      - name: Check cluster access
-        run: kubectl get nodes
-
-  mgmt-cluster-access:
-    runs-on: ubuntu-latest
-    permissions:
-      id-token: 'write'
-      contents: 'read'
-
-    steps:
-      - name: Checkout repo
-        uses: actions/checkout@v4
-
-      # Login to Azure using OIDC (no secret needed if federated identity is set up)
-      - name: Azure Login
-        uses: azure/login@v2
-        with:
-          client-id: ${{ secrets.AZURE_CLIENT_ID }}
-          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
-          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
-          # if not using federated identity:
-          # client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
-
       # Get AKS credentials
       - name: Get AKS credentials
         run: |
+          az aks get-credentials \
+            --resource-group ${{ secrets.SVC_RESOURCE_GROUP }} \
+            --name ${{ secrets.SVC_CLUSTER_NAME }} \
+            --overwrite-existing \
+            -f ./svc-cluster.kubeconfig
           az aks get-credentials \
             --resource-group ${{ secrets.MGMT_RESOURCE_GROUP }} \
             --name ${{ secrets.MGMT_CLUSTER_NAME }} \
-            --overwrite-existing
-
-      # Install kubelogin (needed for AAD-enabled clusters)
-      - name: Install kubelogin
-        run: |
-          az aks install-cli
-          kubelogin --version
+            --overwrite-existing \
+            -f ./mgmt-cluster.kubeconfig
 
       # Convert kubeconfig for non-interactive login (GitHub runner)
       - name: Convert kubeconfig for kubelogin
         run: |
-          kubelogin convert-kubeconfig -l azurecli
+          kubelogin convert-kubeconfig -l azurecli --kubeconfig ./svc-cluster.kubeconfig
+          kubelogin convert-kubeconfig -l azurecli --kubeconfig ./mgmt-cluster.kubeconfig
 
       # Verify kubectl works
       - name: Check cluster access
-        run: kubectl get nodes
+        run: |
+          kubectl --kubeconfig ./svc-cluster.kubeconfig get pods -A -l app=maestro
+          kubectl --kubeconfig ./mgmt-cluster.kubeconfig get pods -A -l app=maestro-agent
+
+      # Setup go
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '>=1.23.0'
+      - name: install ginkgo
+        run: go install github.com/onsi/ginkgo/v2/ginkgo@v2.15.0
+
+      # Running the e2e test
+      - name: e2e test
+        run: |
+          curl -fsSL -O https://github.com/knight42/krelay/releases/download/v0.1.2/kubectl-relay_v0.1.2_linux-amd64.tar.gz
+          tar -xzf kubectl-relay_v0.1.2_linux-amd64.tar.gz kubectl-relay
+          chmod a+x kubectl-relay
+          $(pwd)/kubectl-relay --kubeconfig ./svc-cluster.kubeconfig -n maestro svc/maestro 8000:8000 &
+          $(pwd)/kubectl-relay --kubeconfig ./svc-cluster.kubeconfig -n maestro svc/maestro-grpc 8090:8090 &
```

Comment:
the connection will be denied by istio sidecar.

Reply:
should we have this annotation by default in this long running cluster? or let action to update the value and changed it back after the test done?

Reply:
suggest removing the annotation after the test. We may add other tests (e.g., similar client with cluster-service) that require Istio to intercept the traffic.

Reply:
yes, we have a steps to reset the annotations in line 91